crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -113,13 +113,14 @@ impl Node {
     /// This is mainly used for instant payments where the receiver does not have a lightning
     /// channel yet, e.g. Alice does not have a channel with Bob yet but wants to
     /// receive a LN payment. Clair pays to Bob who opens a channel to Alice and pays her.
-    pub fn create_intercept_scid(&self, target_node: PublicKey) -> u64 {
+    pub fn create_intercept_scid(&self, target_node: PublicKey) -> Result<u64> {
         let intercept_scid = self.channel_manager.get_intercept_scid();
-        self.fake_channel_payments
+        let mut fake_channel_payment = self
+            .fake_channel_payments
             .lock()
-            .unwrap()
-            .insert(intercept_scid, target_node);
-        intercept_scid
+            .map_err(|e| anyhow!("{e:#}"))?;
```

Comment:
🤔 I think we should just `unwrap` or `expect` when dealing with locks. The reason why this panics is because the mutex was poisoned in the event handler that has all those _other_ unwraps. A mutex is poisoned if the guard is held and a panic occurs.